python/cudf/dataframe/dataframe.py

```diff
@@ -468,73 +494,122 @@ def _apply_op(self, fn, other=None):
                 result[col] = getattr(self._cols[col], fn)(other[k])
         elif isinstance(other, DataFrame):
             max_num_rows = max(self.shape[0], other.shape[0])
+
+            def fallback(col, fn):
+                if fill_value is None:
+                    return Series(cudautils.full(max_num_rows,
+                                  np.dtype('float64').type(np.nan),
+                                  'float64'), nan_as_null=False)
```

Comment:
We shouldn't be filling with `NaN`, we should be filling with null, which means setting the relevant bitmask values. This will handle it via an extremely inefficient codepath in filling with NaNs just to convert those back to a bitmask.

Reply:
done